<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>PRONE with Spike-In Data • PRONE</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="PRONE with Spike-In Data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PRONE</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.16</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/PRONE.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Preprocessing.html">Preprocessing</a></li>
    <li><a class="dropdown-item" href="../articles/Normalization.html">Normalization</a></li>
    <li><a class="dropdown-item" href="../articles/Imputation.html">Imputation</a></li>
    <li><a class="dropdown-item" href="../articles/Differential_Expression.html">Differential Expression Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/Spike_In_Data.html">PRONE with Spike-In Data</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/lisiarend/PRONE/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>PRONE with Spike-In Data</h1>
                        <h4 data-toc-skip class="author">Lis Arend</h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/lisiarend/PRONE/blob/HEAD/vignettes/Spike_In_Data.Rmd" class="external-link"><code>vignettes/Spike_In_Data.Rmd</code></a></small>
      <div class="d-none name"><code>Spike_In_Data.Rmd</code></div>
    </div>

    
    
<div class="section level2" number="1">
<h2 id="load-prone-package">
<span class="header-section-number">1</span> Load PRONE Package<a class="anchor" aria-label="anchor" href="#load-prone-package"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lisiarend/PRONE" class="external-link">PRONE</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="2">
<h2 id="example-spike-in-data-set">
<span class="header-section-number">2</span> Example Spike-in Data Set<a class="anchor" aria-label="anchor" href="#example-spike-in-data-set"></a>
</h2>
<p>PRONE also offers some additional functionalities for the evaluation of normalization techniques on spike-in data sets with a known ground truth. These functionalities will be delineated subsequently. Additionally, all functionalities detailed in the context of real-world data sets remain applicable to the SummarizedExperiment object associated with spike-in data sets.</p>
<p>The example spike-in data set is from <span class="citation">(Cox et al. 2014)</span>. For the illustration of the package functionality, a smaller subset of the data consisting of 1500 proteins was used. For the complete data set, please refer to the original publication.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readPRONE_example.html">readPRONE_example</a></span><span class="op">(</span><span class="st">"Ecoli_human_MaxLFQ_protein_intensities.csv"</span><span class="op">)</span></span>
<span><span class="va">md_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readPRONE_example.html">readPRONE_example</a></span><span class="op">(</span><span class="st">"Ecoli_human_MaxLFQ_metadata.csv"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">data_path</span><span class="op">)</span></span>
<span><span class="va">md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">md_path</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="3">
<h2 id="load-data">
<span class="header-section-number">3</span> Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<p>Before loading the data into a SummarizedExperiment object, it is important to check the organism of the protein groups. Proteins should be uniquely assigned to either the spike-in organism or the background organism. If some protein groups are mixed, they should be removed from the data since these can’t be used for classification into true positives, false positives, etc.</p>
<p>In our example, we can extract the information from the “Fasta.headers” column:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check if some protein groups are mixed</span></span>
<span><span class="va">mixed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"Homo sapiens.*Escherichia|Escherichia.*Homo sapiens"</span>,<span class="va">data</span><span class="op">$</span><span class="va">Fasta.headers</span><span class="op">)</span> </span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">[</span><span class="op">!</span><span class="va">mixed</span>,<span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">mixed</span><span class="op">)</span> </span>
<span><span class="co">#&gt; mixed</span></span>
<span><span class="co">#&gt; FALSE </span></span>
<span><span class="co">#&gt;  1000</span></span></code></pre></div>
<p>In this case, all proteins were assigned either as a spike-in or background protein. Hence, a SummarizedExperiment container of the data can be created. However, before we need to add a column of the actual organism that can be used to calculate true positives, false positives, etc.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span><span class="op">$</span><span class="va">Spiked</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"HUMAN"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">data</span><span class="op">$</span><span class="va">Spiked</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"ECOLI"</span>, <span class="va">data</span><span class="op">$</span><span class="va">Fasta.headers</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"ECOLI"</span></span></code></pre></div>
<p>In contrast to the real-world data sets, you need to specify the “spike_column”, “spike_value”, “spike_concentration”, and utilize the <code><a href="../reference/load_spike_data.html">load_spike_data()</a></code> function for this purpose.</p>
<p>Here, the “spike_column” denotes the column in the protein intensity data file encompassing information whether a proteins is classified as spike-in or background protein. The “spike_value” determines the value/identifier that is used to identify the spike-in proteins in the “spike_column”, and the “spike_concentration” identifies the column containing the concentration values of the spike-in proteins (in this case the different conditions that will be tested in DE analysis).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_spike_data.html">load_spike_data</a></span><span class="op">(</span><span class="va">data</span>, </span>
<span>                      <span class="va">md</span>, </span>
<span>                      spike_column <span class="op">=</span> <span class="st">"Spiked"</span>, </span>
<span>                      spike_value <span class="op">=</span> <span class="st">"ECOLI"</span>, </span>
<span>                      spike_concentration <span class="op">=</span> <span class="st">"Concentration"</span>,</span>
<span>                      protein_column <span class="op">=</span> <span class="st">"Protein.IDs"</span>, </span>
<span>                      gene_column <span class="op">=</span> <span class="st">"Gene.names"</span>, </span>
<span>                      ref_samples <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                      batch_column <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                      condition_column <span class="op">=</span> <span class="st">"Condition"</span>, </span>
<span>                      label_column <span class="op">=</span> <span class="st">"Label"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2" number="4">
<h2 id="overview-of-the-data">
<span class="header-section-number">4</span> Overview of the Data<a class="anchor" aria-label="anchor" href="#overview-of-the-data"></a>
</h2>
<p>To get an overview on the number of identified (non-NA) spike-in proteins per sample, you can use the <code><a href="../reference/plot_identified_spiked_proteins.html">plot_identified_spiked_proteins()</a></code> function.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_identified_spiked_proteins.html">plot_identified_spiked_proteins</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-6"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-6-1.png" alt="Overview of the number of identified spike-in proteins per sample, colored by condition. In this data set, the conditions were labeled with H and L. H indicates the sample group with high concentrations of spike-in proteins added to the background proteome, while L represents the sample group with low concentrations." width="768"><p class="caption">
Figure 4.1: Overview of the number of identified spike-in proteins per sample, colored by condition. In this data set, the conditions were labeled with H and L. H indicates the sample group with high concentrations of spike-in proteins added to the background proteome, while L represents the sample group with low concentrations.
</p>
</div>
<p>To compare the distributions of spike-in and human proteins in the different sample groups (here high-low), use the function <code><a href="../reference/plot_histogram_spiked.html">plot_histogram_spiked()</a></code>. Again, “condition = NULL” means that the condition specified by loading the data is used, but you can also specify any other column of the meta data.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_histogram_spiked.html">plot_histogram_spiked</a></span><span class="op">(</span><span class="va">se</span>, condition <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-7"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-7-1.png" alt="Histogram of the protein intensities of the spike-in proteins (ECOLI) and the background proteins (HUMAN) in the different conditions. This plot helps to compare the distributions of spike-in (red) and background proteins (grey) for the different spike-in levels." width="768"><p class="caption">
Figure 4.2: Histogram of the protein intensities of the spike-in proteins (ECOLI) and the background proteins (HUMAN) in the different conditions. This plot helps to compare the distributions of spike-in (red) and background proteins (grey) for the different spike-in levels.
</p>
</div>
<p>If you want to have a look at the amount of actual measure spike-in, you can use the <code><a href="../reference/plot_profiles_spiked.html">plot_profiles_spiked()</a></code> function. Moreover, you can analyze whether the intensities of the background proteins, here HUMAN proteins, are constant across the different spike-in concentrations.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_profiles_spiked.html">plot_profiles_spiked</a></span><span class="op">(</span><span class="va">se</span>, xlab <span class="op">=</span> <span class="st">"Concentration"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-8"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-8-1.png" alt="Profiles of the spike-in proteins (ECOLI) and the background proteins (HUMAN) in the different conditions. This plot helps to analyze whether the intensities of the background proteins are constant across the different spike-in concentrations. Spike-in proteins (red) should increase in intensity with increasing spike-in concentrations, while background proteins (grey) should remain constant." width="768"><p class="caption">
Figure 4.3: Profiles of the spike-in proteins (ECOLI) and the background proteins (HUMAN) in the different conditions. This plot helps to analyze whether the intensities of the background proteins are constant across the different spike-in concentrations. Spike-in proteins (red) should increase in intensity with increasing spike-in concentrations, while background proteins (grey) should remain constant.
</p>
</div>
</div>
<div class="section level2" number="5">
<h2 id="preprocessing-normalization-imputation">
<span class="header-section-number">5</span> Preprocessing, Normalization, &amp; Imputation<a class="anchor" aria-label="anchor" href="#preprocessing-normalization-imputation"></a>
</h2>
<p>Given that the preprocessing (including filtering of proteins and samples), normalization, and imputation operations remain invariant for spike-in data sets compared to real-world data sets, the same methodologies can be employed across both types. In this context, normalization will only be demonstrated here as the performance of the methods will be evaluated in DE analysis, while detailed descriptions of the other functionalities are available in preceding sections.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_se.html">normalize_se</a></span><span class="op">(</span><span class="va">se</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span>, <span class="st">"LoessF"</span><span class="op">)</span>, </span>
<span>                        combination_pattern <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Median completed.</span></span>
<span><span class="co">#&gt; Mean completed.</span></span>
<span><span class="co">#&gt; MAD completed.</span></span>
<span><span class="co">#&gt; LoessF completed.</span></span></code></pre></div>
</div>
<div class="section level2" number="6">
<h2 id="differential-expression-analysis">
<span class="header-section-number">6</span> Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a>
</h2>
<p>Due to the known spike-in concentrations, the normalization methods can be evaluated based on their ability to detect DE proteins. The DE analysis can be conducted using the same methodology as for real-world data sets. However, other visualization options are available and performance metrics can be calculated for spike-in data sets.</p>
<div class="section level3" number="6.1">
<h3 id="run-de-analysis">
<span class="header-section-number">6.1</span> Run DE Analysis<a class="anchor" aria-label="anchor" href="#run-de-analysis"></a>
</h3>
<p>First, you need to specify the comparisons you want to perform in DE analysis. For this, a special function was developed which helps to build the right comparison strings.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comparisons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/specify_comparisons.html">specify_comparisons</a></span><span class="op">(</span><span class="va">se_norm</span>, condition <span class="op">=</span> <span class="st">"Condition"</span>, </span>
<span>                                   sep <span class="op">=</span> <span class="cn">NULL</span>, control <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p>Then you can run DE analysis:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">de_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_DE.html">run_DE</a></span><span class="op">(</span>se <span class="op">=</span> <span class="va">se_norm</span>, </span>
<span>                     comparisons <span class="op">=</span> <span class="va">comparisons</span>,</span>
<span>                     ain <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     condition <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     DE_method <span class="op">=</span> <span class="st">"limma"</span>, </span>
<span>                     covariate <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     logFC <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     logFC_up <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                     logFC_down <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, </span>
<span>                     p_adj <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     alpha <span class="op">=</span> <span class="fl">0.05</span>, </span>
<span>                     B <span class="op">=</span> <span class="fl">100</span>, </span>
<span>                     K <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; DE Analysis will not be performed on raw data.</span></span>
<span><span class="co">#&gt; log2: DE analysis completed.</span></span>
<span><span class="co">#&gt; Median: DE analysis completed.</span></span>
<span><span class="co">#&gt; Mean: DE analysis completed.</span></span>
<span><span class="co">#&gt; MAD: DE analysis completed.</span></span>
<span><span class="co">#&gt; LoessF: DE analysis completed.</span></span></code></pre></div>
</div>
<div class="section level3" number="6.2">
<h3 id="evaluate-de-results-with-performance-metrics">
<span class="header-section-number">6.2</span> Evaluate DE Results with Performance Metrics<a class="anchor" aria-label="anchor" href="#evaluate-de-results-with-performance-metrics"></a>
</h3>
<p>Before being able to visualize the DE results, you need to run <code><a href="../reference/get_spiked_stats_DE.html">get_spiked_stats_DE()</a></code> to calculate multiple performance metrics, such as the number of true positives, number fo false positives, etc.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_spiked_stats_DE.html">get_spiked_stats_DE</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Show tp, fp, fn, tn, with F1 score</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">stats</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Assay"</span>, <span class="st">"Comparison"</span>, <span class="st">"TP"</span>, <span class="st">"FP"</span>, <span class="st">"FN"</span>, <span class="st">"TN"</span>, <span class="st">"F1Score"</span><span class="op">)</span><span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Performance metrics for the different normalization methods and pairwise comparisons."</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
<span id="tab:unnamed-chunk-12">Table 6.1: </span>Performance metrics for the different normalization methods and pairwise comparisons.</caption>
<thead><tr class="header">
<th align="left">Assay</th>
<th align="left">Comparison</th>
<th align="right">TP</th>
<th align="right">FP</th>
<th align="right">FN</th>
<th align="right">TN</th>
<th align="right">F1Score</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">LoessF</td>
<td align="left">H-L</td>
<td align="right">169</td>
<td align="right">26</td>
<td align="right">93</td>
<td align="right">712</td>
<td align="right">0.74</td>
</tr>
<tr class="even">
<td align="left">MAD</td>
<td align="left">H-L</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">262</td>
<td align="right">737</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td align="left">Mean</td>
<td align="left">H-L</td>
<td align="right">138</td>
<td align="right">62</td>
<td align="right">124</td>
<td align="right">676</td>
<td align="right">0.60</td>
</tr>
<tr class="even">
<td align="left">Median</td>
<td align="left">H-L</td>
<td align="right">156</td>
<td align="right">41</td>
<td align="right">106</td>
<td align="right">697</td>
<td align="right">0.68</td>
</tr>
<tr class="odd">
<td align="left">log2</td>
<td align="left">H-L</td>
<td align="right">162</td>
<td align="right">40</td>
<td align="right">100</td>
<td align="right">698</td>
<td align="right">0.70</td>
</tr>
</tbody>
</table>
<p>You have different options to visualize the amount of true and false positives for the different normalization techniques and pairwise comparisons. For all these functions, you can specify a subset of normalization methods and comparisons. If you do not specify anything, all normalization methods and comparisons of the “stats” data frame are used.</p>
<p>The <code><a href="../reference/plot_TP_FP_spiked_bar.html">plot_TP_FP_spiked_bar()</a></code> function generates a barplot showing the number of false positives and true positives for each normalization method and is facetted by pairwise comparison.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_TP_FP_spiked_bar.html">plot_TP_FP_spiked_bar</a></span><span class="op">(</span><span class="va">stats</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span>, <span class="st">"LoessF"</span><span class="op">)</span>, </span>
<span>                      comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of stats will be visualized.</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-13"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-13-1.png" alt="Barplot showing the number of false positives (FP) and true positives (TP) for each normalization method and is facetted by pairwise comparison. This plot shows the impact of normalization on DE results." width="768"><p class="caption">
Figure 6.1: Barplot showing the number of false positives (FP) and true positives (TP) for each normalization method and is facetted by pairwise comparison. This plot shows the impact of normalization on DE results.
</p>
</div>
<p>If many pairwise comparisons were performed, the <code><a href="../reference/plot_TP_FP_spiked_box.html">plot_TP_FP_spiked_box()</a></code> function can be used to visualize the distribution of true and false positives for all pairwise comparisons in a boxplot.</p>
<p>Given that the data set encompasses merely two distinct spike-in concentrations and therefore only one pairwise comparison has been conducted in DE analysis, a barplot would be more appropriate. Nonetheless, for demonstration purposes, a boxplot will be used here.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_TP_FP_spiked_box.html">plot_TP_FP_spiked_box</a></span><span class="op">(</span><span class="va">stats</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span>, <span class="st">"LoessF"</span><span class="op">)</span>, </span>
<span>                      comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of stats will be visualized.</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-14"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-14-1.png" alt="Boxplot showing the distribution of true positives (TP) and false positives (FP) for each normalization method across all pairwise comparisons. This plot helps to visualize the distribution of TP and FP for each normalization method. Since in this dataset, only two conditions are available, hence a single pairwise comparison, this plot is not very informative." width="768"><p class="caption">
Figure 6.2: Boxplot showing the distribution of true positives (TP) and false positives (FP) for each normalization method across all pairwise comparisons. This plot helps to visualize the distribution of TP and FP for each normalization method. Since in this dataset, only two conditions are available, hence a single pairwise comparison, this plot is not very informative.
</p>
</div>
<p>Furthermore, similarly the <code><a href="../reference/plot_TP_FP_spiked_scatter.html">plot_TP_FP_spiked_scatter()</a></code> function can be used to visualize the true and false positives in a scatter plot. Here, a scatterplot of the median true positives and false positives is calculated across all comparisons and displayed for each normalization method with errorbars showing the Q1 and Q3. Here again, this plot is more suitable for data sets with multiple pairwise comparisons.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_TP_FP_spiked_scatter.html">plot_TP_FP_spiked_scatter</a></span><span class="op">(</span><span class="va">stats</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of stats will be visualized.</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-15"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-15-1.png" alt="Scatter plot showing the median true positives (TP) and false positives (FP) for each normalization method across all pairwise comparisons." width="768"><p class="caption">
Figure 6.3: Scatter plot showing the median true positives (TP) and false positives (FP) for each normalization method across all pairwise comparisons.
</p>
</div>
<p>Furthermore, other performance metrics that the number of true positives and false positives can be visualized using the <code><a href="../reference/plot_stats_spiked_heatmap.html">plot_stats_spiked_heatmap()</a></code> function. Here, currently, the sensitivity, specificity, precision, FPR, F1 score, and accuracy can be visualized for the different normalization methods and pairwise comparisons.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_stats_spiked_heatmap.html">plot_stats_spiked_heatmap</a></span><span class="op">(</span><span class="va">stats</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, </span>
<span>                          metrics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Precision"</span>, <span class="st">"F1Score"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of stats will be visualized.</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-16"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-16-1.png" alt="Heatmap showing a selection of performance metrics for the different normalization methods and pairwise comparisons." width="768"><p class="caption">
Figure 6.4: Heatmap showing a selection of performance metrics for the different normalization methods and pairwise comparisons.
</p>
</div>
<p>Finally, receiver operating characteristic (ROC) curves and AUC values can be calculated and visualized using the <code><a href="../reference/plot_ROC_AUC_spiked.html">plot_ROC_AUC_spiked()</a></code> function. This function returns a plot showing the ROC curves, a bar plot with the AUC values for each comparison, and a boxplot with the distributions of AUC values across all comparisons.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_ROC_AUC_spiked.html">plot_ROC_AUC_spiked</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"LoessF"</span><span class="op">)</span>, </span>
<span>                    comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span>
<span><span class="co">#&gt; $ROC</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-17-1"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-17-1.png" alt="ROC curves and AUC barplots and boxplots for the different normalization methods and pairwise comparisons. AUC barplots are shown for each pairwise comparison, while the boxplot shows the distribution of AUC values across all comparisons." width="768"><p class="caption">
Figure 6.5: ROC curves and AUC barplots and boxplots for the different normalization methods and pairwise comparisons. AUC barplots are shown for each pairwise comparison, while the boxplot shows the distribution of AUC values across all comparisons.
</p>
</div>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $AUC_bars</span></span></code></pre>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-17-2"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-17-2.png" alt="ROC curves and AUC barplots and boxplots for the different normalization methods and pairwise comparisons. AUC barplots are shown for each pairwise comparison, while the boxplot shows the distribution of AUC values across all comparisons." width="768"><p class="caption">
Figure 6.6: ROC curves and AUC barplots and boxplots for the different normalization methods and pairwise comparisons. AUC barplots are shown for each pairwise comparison, while the boxplot shows the distribution of AUC values across all comparisons.
</p>
</div>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $AUC_box</span></span></code></pre>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-17-3"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-17-3.png" alt="ROC curves and AUC barplots and boxplots for the different normalization methods and pairwise comparisons. AUC barplots are shown for each pairwise comparison, while the boxplot shows the distribution of AUC values across all comparisons." width="768"><p class="caption">
Figure 6.7: ROC curves and AUC barplots and boxplots for the different normalization methods and pairwise comparisons. AUC barplots are shown for each pairwise comparison, while the boxplot shows the distribution of AUC values across all comparisons.
</p>
</div>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $AUC_dt</span></span>
<span><span class="co">#&gt;   PANEL Comparison group  Assay       AUC</span></span>
<span><span class="co">#&gt; 1     1        H-L     1 LoessF 0.8654546</span></span>
<span><span class="co">#&gt; 2     1        H-L     2   Mean 0.7693012</span></span>
<span><span class="co">#&gt; 3     1        H-L     3 Median 0.8638456</span></span></code></pre>
</div>
<div class="section level3" number="6.3">
<h3 id="log-fold-change-distributions">
<span class="header-section-number">6.3</span> Log Fold Change Distributions<a class="anchor" aria-label="anchor" href="#log-fold-change-distributions"></a>
</h3>
<p>Furthermore, you can visualize the distribution of log fold changes for the different conditions using the <code><a href="../reference/plot_fold_changes_spiked.html">plot_fold_changes_spiked()</a></code> function. The fold changes of the background proteins should be centered around zero, while the spike-in proteins should be centered around the actual log fold change calculated based on the spike-in concentrations.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_fold_changes_spiked.html">plot_fold_changes_spiked</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span>, condition <span class="op">=</span> <span class="st">"Condition"</span>, </span>
<span>                         ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-18"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-18-1.png" alt="Boxplot showing the distribution of log fold changes for the different normalization methods and pairwise comparisons. The dotted blue line is at y = 0 because logFC values of the background proteins should be centered around 0, while the dotted red line shows the expected logFC value based on the spike-in concentrations of both sample groups." width="768"><p class="caption">
Figure 6.8: Boxplot showing the distribution of log fold changes for the different normalization methods and pairwise comparisons. The dotted blue line is at y = 0 because logFC values of the background proteins should be centered around 0, while the dotted red line shows the expected logFC value based on the spike-in concentrations of both sample groups.
</p>
</div>
</div>
<div class="section level3" number="6.4">
<h3 id="p-value-distributions">
<span class="header-section-number">6.4</span> P-Value Distributions<a class="anchor" aria-label="anchor" href="#p-value-distributions"></a>
</h3>
<p>Similarly, the distributions of the p-values can be visualized using the <code><a href="../reference/plot_pvalues_spiked.html">plot_pvalues_spiked()</a></code> function.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_pvalues_spiked.html">plot_pvalues_spiked</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, </span>
<span>                    comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-19"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-19-1.png" alt="Boxplot showing the distribution of p-values for the different normalization methods and pairwise comparisons." width="768"><p class="caption">
Figure 6.9: Boxplot showing the distribution of p-values for the different normalization methods and pairwise comparisons.
</p>
</div>
</div>
<div class="section level3" number="6.5">
<h3 id="log-fold-change-thresholds">
<span class="header-section-number">6.5</span> Log Fold Change Thresholds<a class="anchor" aria-label="anchor" href="#log-fold-change-thresholds"></a>
</h3>
<p>Due to the high amount of false positives encountered in spike-in data sets, we provided a function to test for different log fold change thresholds to try to reduce the amount of false positives. The function <code><a href="../reference/plot_logFC_thresholds_spiked.html">plot_logFC_thresholds_spiked()</a></code> can be used to visualize the number of true positives and false positives for different log fold change thresholds.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_logFC_thresholds_spiked.html">plot_logFC_thresholds_spiked</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                                      ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, </span>
<span>                                      nrow <span class="op">=</span> <span class="fl">1</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-21"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-21-1.png" alt="Barplot showing the number of true positives for each normalization method and pairwise comparison for different log fold change thresholds. This plot helps to analyze the impact of different log fold change thresholds on the number of true positives. The dotted line in the plot shows the expected logFC value based on the spike-in concentrations of both sample groups." width="768"><p class="caption">
Figure 6.10: Barplot showing the number of true positives for each normalization method and pairwise comparison for different log fold change thresholds. This plot helps to analyze the impact of different log fold change thresholds on the number of true positives. The dotted line in the plot shows the expected logFC value based on the spike-in concentrations of both sample groups.
</p>
</div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:unnamed-chunk-22"></span>
<img src="Spike_In_Data_files/figure-html/unnamed-chunk-22-1.png" alt="Barplot showing the number of false positives for each normalization method and pairwise comparison for different log fold change thresholds. This plot helps to analyze the impact of different log fold change thresholds on the number of false positives. The dotted line in the plot shows the expected logFC value based on the spike-in concentrations of both sample groups." width="768"><p class="caption">
Figure 6.11: Barplot showing the number of false positives for each normalization method and pairwise comparison for different log fold change thresholds. This plot helps to analyze the impact of different log fold change thresholds on the number of false positives. The dotted line in the plot shows the expected logFC value based on the spike-in concentrations of both sample groups.
</p>
</div>
</div>
</div>
<div class="section level2" number="7">
<h2 id="session-info">
<span class="header-section-number">7</span> Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Berlin</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices datasets  utils     methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] PRONE_0.99.16               SummarizedExperiment_1.34.0</span></span>
<span><span class="co">#&gt;  [3] Biobase_2.64.0              GenomicRanges_1.56.1       </span></span>
<span><span class="co">#&gt;  [5] GenomeInfoDb_1.40.1         IRanges_2.38.1             </span></span>
<span><span class="co">#&gt;  [7] S4Vectors_0.42.1            BiocGenerics_0.50.0        </span></span>
<span><span class="co">#&gt;  [9] MatrixGenerics_1.16.0       matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] rlang_1.1.4                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [3] clue_0.3-65                 compiler_4.4.1             </span></span>
<span><span class="co">#&gt;   [5] vctrs_0.6.5                 reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;   [7] stringr_1.5.1               ProtGenerics_1.36.0        </span></span>
<span><span class="co">#&gt;   [9] pkgconfig_2.0.3             crayon_1.5.3               </span></span>
<span><span class="co">#&gt;  [11] fastmap_1.2.0               XVector_0.44.0             </span></span>
<span><span class="co">#&gt;  [13] labeling_0.4.3              utf8_1.2.4                 </span></span>
<span><span class="co">#&gt;  [15] rmarkdown_2.27              UCSC.utils_1.0.0           </span></span>
<span><span class="co">#&gt;  [17] preprocessCore_1.66.0       purrr_1.0.2                </span></span>
<span><span class="co">#&gt;  [19] xfun_0.46                   MultiAssayExperiment_1.30.3</span></span>
<span><span class="co">#&gt;  [21] zlibbioc_1.50.0             cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [23] jsonlite_1.8.8              highr_0.11                 </span></span>
<span><span class="co">#&gt;  [25] DelayedArray_0.30.1         BiocParallel_1.38.0        </span></span>
<span><span class="co">#&gt;  [27] parallel_4.4.1              cluster_2.1.6              </span></span>
<span><span class="co">#&gt;  [29] R6_2.5.1                    RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;  [31] bslib_0.8.0                 stringi_1.8.4              </span></span>
<span><span class="co">#&gt;  [33] limma_3.60.4                jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [35] Rcpp_1.0.13                 bookdown_0.40              </span></span>
<span><span class="co">#&gt;  [37] iterators_1.0.14            knitr_1.48                 </span></span>
<span><span class="co">#&gt;  [39] splines_4.4.1               Matrix_1.7-0               </span></span>
<span><span class="co">#&gt;  [41] igraph_2.0.3                tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [43] rstudioapi_0.16.0           abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [45] yaml_2.3.10                 doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [47] ggtext_0.1.2                codetools_0.2-20           </span></span>
<span><span class="co">#&gt;  [49] affy_1.82.0                 lattice_0.22-6             </span></span>
<span><span class="co">#&gt;  [51] tibble_3.2.1                plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;  [53] withr_3.0.1                 evaluate_0.24.0            </span></span>
<span><span class="co">#&gt;  [55] xml2_1.3.6                  pillar_1.9.0               </span></span>
<span><span class="co">#&gt;  [57] affyio_1.74.0               BiocManager_1.30.23        </span></span>
<span><span class="co">#&gt;  [59] renv_1.0.7                  foreach_1.5.2              </span></span>
<span><span class="co">#&gt;  [61] MSnbase_2.30.1              MALDIquant_1.22.2          </span></span>
<span><span class="co">#&gt;  [63] ncdf4_1.22                  generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [65] ggplot2_3.5.1               munsell_0.5.1              </span></span>
<span><span class="co">#&gt;  [67] scales_1.3.0                glue_1.7.0                 </span></span>
<span><span class="co">#&gt;  [69] lazyeval_0.2.2              tools_4.4.1                </span></span>
<span><span class="co">#&gt;  [71] data.table_1.15.4           mzID_1.42.0                </span></span>
<span><span class="co">#&gt;  [73] QFeatures_1.14.2            vsn_3.72.0                 </span></span>
<span><span class="co">#&gt;  [75] mzR_2.38.0                  XML_3.99-0.17              </span></span>
<span><span class="co">#&gt;  [77] grid_4.4.1                  impute_1.78.0              </span></span>
<span><span class="co">#&gt;  [79] tidyr_1.3.1                 MsCoreUtils_1.16.1         </span></span>
<span><span class="co">#&gt;  [81] colorspace_2.1-1            GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">#&gt;  [83] PSMatch_1.8.0               cli_3.6.3                  </span></span>
<span><span class="co">#&gt;  [85] plotROC_2.3.1               fansi_1.0.6                </span></span>
<span><span class="co">#&gt;  [87] S4Arrays_1.4.1              dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [89] AnnotationFilter_1.28.0     pcaMethods_1.96.0          </span></span>
<span><span class="co">#&gt;  [91] gtable_0.3.5                sass_0.4.9                 </span></span>
<span><span class="co">#&gt;  [93] digest_0.6.36               SparseArray_1.4.8          </span></span>
<span><span class="co">#&gt;  [95] farver_2.1.2                htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt;  [97] htmltools_0.5.8.1           lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt;  [99] httr_1.4.7                  NormalyzerDE_1.22.0        </span></span>
<span><span class="co">#&gt; [101] statmod_1.5.0               gridtext_0.1.5             </span></span>
<span><span class="co">#&gt; [103] MASS_7.3-61</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-cox_accurate_2014" class="csl-entry">
Cox, Jürgen, Marco Y. Hein, Christian A. Luber, Igor Paron, Nagarjuna Nagaraj, and Matthias Mann. 2014. <span>“Accurate Proteome-Wide Label-Free Quantification by Delayed Normalization and Maximal Peptide Ratio Extraction, Termed Maxlfq.”</span> <em>Molecular &amp; Cellular Proteomics</em> 13 (9): 2513–26. <a href="https://doi.org/10.1074/mcp.M113.031591" class="external-link">https://doi.org/10.1074/mcp.M113.031591</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Lis Arend.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
